server/src/main/java/org/apache/druid/server/DruidInternalDynamicConfigResource.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.server;
+
+import com.google.inject.Inject;
+import com.sun.jersey.spi.container.ResourceFilters;
+import org.apache.druid.client.CoordinatorDynamicConfigView;
+import org.apache.druid.server.coordinator.CoordinatorDynamicConfig;
+import org.apache.druid.server.http.security.ConfigResourceFilter;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Path("/druid-internal/v1/dynamicConfiguration")
+public class DruidInternalDynamicConfigResource
+{
+  private final CoordinatorDynamicConfigView coordinatorDynamicConfigView;
+
+  @Inject
+  public DruidInternalDynamicConfigResource(CoordinatorDynamicConfigView coordinatorDynamicConfigView)
+  {
+    this.coordinatorDynamicConfigView = coordinatorDynamicConfigView;
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  @ResourceFilters(ConfigResourceFilter.class)
+  @Path("/coordinatorDynamicConfig")
+  public Response getDynamicConfig()
+  {
+    return Response.ok(coordinatorDynamicConfigView.getDynamicConfiguration()).build();
+  }
+
+  @POST
+  @Consumes(MediaType.APPLICATION_JSON)
+  @ResourceFilters(ConfigResourceFilter.class)
+  @Path("/coordinatorDynamicConfig")
+  public Response setDynamicConfig(final CoordinatorDynamicConfig dynamicConfig)
+  {
+    coordinatorDynamicConfigView.setDynamicConfiguration(dynamicConfig);
+    return Response.ok("OK").build();
```

Comment:
Nicer to include a proper JSON response, even if it is just `{"success": true}`.
```suggestion
    return Response.ok("OK").build();
```

Reply:
Dropped the "OK"